ui/packages/catalog-ui-search/src/main/webapp/component/navigation-middle/navigation-middle.view.js

```diff
@@ -14,17 +14,31 @@
  **/
 /*global require*/
 var Marionette = require('marionette')
-var template = require('./navigation-middle.hbs')
 var CustomElements = require('../../js/CustomElements.js')
-const router = require('../router/router.js')
+const properties = require('properties')
+import * as React from 'react'
+import { IntlProvider, FormattedMessage } from 'react-intl'
 
 module.exports = Marionette.LayoutView.extend({
-  template: template,
+  template(props) {
+    return (
+      <div className={props.menuClass}>
+        <IntlProvider locale={navigator.language}>
+          <FormattedMessage
+            id="sourcesKeyword"
+            defaultMessage={props.menuText}
+          />
+        </IntlProvider>
+      </div>
+    )
+  },
   tagName: CustomElements.register('navigation-middle'),
   serializeData: function() {
+    const menuText = properties.keywords[this.options.text] || this.options.text
```

Comment:
❗️ This couples the English value to the i18n key.  This will not work well when we start to translate long messages that could be multiple sentences.  Options should pass in a key and/or text.  Only the key should be used to lookup messages for the configured locale.

Reply:
The intent of this PR was not to add i18n support.  I am ok with us adding a comment where appropriate to explain the current limitations so it will be easier for someone in the future to add full i18n support.

Reply:
Comment added